apps/meteor/app/lib/server/functions/sendMessage.ts

```diff
@@ -215,11 +216,46 @@ export function prepareMessageObject(
 /**
  * Validates and sends the message object.
  */
-export const sendMessage = async function (user: any, message: any, room: any, upsert = false, previewUrls?: string[]) {
+export const sendMessage = async function (
+	user: any,
+	message: any,
+	room: any,
+	upsert = false,
+	previewUrls?: string[],
+	uploadIdsToConfirm?: string[],
+) {
 	if (!user || !message || !room._id) {
 		return false;
 	}
 
+	if (uploadIdsToConfirm !== undefined) {
+		const filesToConfirm: Partial<IUpload>[] = await Promise.all(
+			uploadIdsToConfirm.map(async (fileid) => {
+				const file = await Uploads.findOneById(fileid);
+				if (!file) {
+					throw new Meteor.Error('invalid-file');
+				}
+				return file;
+			}),
+		);
+		if (message?.t !== 'e2e') {
+			const messageFiles: FileProp[] = [];
+			const messageAttachments: MessageAttachment[] = [];
+			const fileAttachmentPromises = filesToConfirm.map(async (file) => {
+				const roomId = message.rid;
+				const { files, attachments } = await parseFileIntoMessageAttachments(file, roomId, user);
+				return { files, attachments };
+			});
+			const filesAndAttachments = await Promise.all(fileAttachmentPromises);
+			filesAndAttachments.forEach(({ files, attachments }) => {
+				messageFiles.push(...files);
+				messageAttachments.push(...attachments);
+			});
+			message.files = messageFiles;
+			message.attachments = messageAttachments;
```

Comment:
This logic can be extracted to a function (maybe named `parseFilesIntoMessageAttachments`)

Reply:
done sir added new function `parseFilesIntoMessageAttachments`